hypershift-operator/controllers/hostedcluster/internal/platform/core.go

```diff
@@ -0,0 +1,60 @@
+package platform
+
+import (
+	"context"
+	"fmt"
+
+	hyperv1 "github.com/openshift/hypershift/api/v1alpha1"
+	"github.com/openshift/hypershift/support/upsert"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	hostedClusterAnnotation = "hypershift.openshift.io/cluster"
+)
+
+type Platform interface {
+	// ReconcileCAPIInfraCR is responsible for creating or updating a CAPIInfraCR.
+	ReconcileCAPIInfraCR(hcluster *hyperv1.HostedCluster, name, controlPlaneNamespace string, apiEndpoint hyperv1.APIEndpoint,
+		c client.Client, ctx context.Context) (client.Object, error)
+
+	// CAPIProviderDeploymentSpec should return a CAPIProvider DeploymentSpec.
+	CAPIProviderDeploymentSpec(controlPlaneNamespace, tokenMinterImage string) (*appsv1.DeploymentSpec, error)
+
+	// HCPPlatformSpec should return a hyperv1.PlatformSpec for the HCP.
+	HCPPlatformSpec(hcp *hyperv1.PlatformSpec, controlPlaneNamespace string) error
+
+	// ReconcileCredentials is responsible for creating or updating resources related to cloud credentials.
+	ReconcileCredentials(c client.Client, ctx context.Context, createOrUpdate upsert.CreateOrUpdateFN,
+		hcluster *hyperv1.HostedCluster,
+		controlPlaneNamespace string) error
+
+	// ReconcileSecretEncryption is responsible for creating or updating resources related to Secrets encryption.
+	ReconcileSecretEncryption(hcluster *hyperv1.HostedCluster, controlPlaneNamespace string, ctx context.Context, c client.Client,
```

Comment:
I wonder if instead of having a separate secret encryption function if this can be consolidated with the ReconcileCredentials function.

Reply:
(Same feedback as in https://github.com/openshift/hypershift/pull/719/files#r758638808 applies here)

Reply:
>I wonder if instead of having a separate secret encryption function if this can be consolidated with the ReconcileCredentials function.

I think a separate func make sense here since this is in hand with a different API field. We can always iterate later.